### PR TITLE
refactor: disallow direct access to attributes in NetworkEventManager

### PR DESCRIPTION
**Pull request by OrKoN**

### CODE REVIEW

**Comment by OrKoN** [comment]
cc @benallfree

File: src/common/NetworkEventManager.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { Protocol } from 'devtools-protocol';
 import { HTTPRequest } from './HTTPRequest.js';
 
-export type QueuedEvents = {
+export type QueuedEventGroup = {
   responseReceived: Protocol.Network.ResponseReceivedEvent;
   promise: Promise<void>;
   resolver: () => void;
```
Comment (benallfree) [inline src/common/NetworkEventManager.ts:7]: From what I could find, `promise` and `resolver` are initialized but never actually awaited or referenced anywhere. I think these fields and the corresponding logic in `NetworkManager` can be removed.

File: src/common/NetworkEventManager.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { Protocol } from 'devtools-protocol';
 import { HTTPRequest } from './HTTPRequest.js';
 
-export type QueuedEvents = {
+export type QueuedEventGroup = {
   responseReceived: Protocol.Network.ResponseReceivedEvent;
```
Comment (benallfree) [inline src/common/NetworkEventManager.ts]: Naming idea: `responseReceivedEvent`, `loadingFinishedEvent`, and `loadingFailedEvent`. Call the whole thing a `QueuedResponseEventGroup`

**Review by benallfree — APPROVED** [review]
lgtm, see other notes or merge as is 👍 

Nice work. I went too far in my encapsulation attempt 😓

File: src/common/NetworkEventManager.ts (hunk 1 above)
Comment (OrKoN) [reply src/common/NetworkEventManager.ts:7]: I think both are used in NetworkManager.

File: src/common/NetworkEventManager.ts [hunk 3]
```diff
@@ -12,10 +12,11 @@ export type QueuedEvents = {
 export type FetchRequestId = string;
 export type NetworkRequestId = string;
 
-export type RedirectInfoMap = Array<{
+type RedirectInfo = {
   event: Protocol.Network.RequestWillBeSentEvent;
   fetchRequestId?: FetchRequestId;
-}>;
+};
+export type RedirectInfoMap = RedirectInfo[];
```
Comment (johanbay) [inline src/common/NetworkEventManager.ts]: If we are renaming types anyway, should we change the name of this type as well?
```suggestion
export type RedirectInfoList = RedirectInfo[];
```

File: src/common/NetworkEventManager.ts [hunk 4]
```diff
@@ -79,12 +80,12 @@ export class NetworkEventManager {
     Protocol.Network.ResponseReceivedExtraInfoEvent[]
   >();
   private _queuedRedirectInfoMap = new Map<NetworkRequestId, RedirectInfoMap>();
```
Comment (johanbay) [inline src/common/NetworkEventManager.ts]:
```suggestion
  private _queuedRedirectInfoMap = new Map<NetworkRequestId, RedirectInfoList>();
```

**Review by johanbay — APPROVED** [review]
LGTM

**Pull request merged**